### DISCUSSION TITLE
Add order page tab programmatically

### DISCUSSION THREAD

**Discussion by piotrpog** [craftcms/commerce#2437] [Ideas]
It would be useful if we could add tabs to the order page programmatically. I know that we can add tabs using field layout, but i would like to have the ability to add them using a plugin.

Use case: I am creating shipping service integration and i would like to have an interface in a separate tab.

**Comment by michaelrog** [comment]
I accomplish this by hijacking the `cp.commerce.order.edit.main-pane` hook, among others.

c.f. https://craftcms.com/docs/commerce/3.x/extend/template-hooks.html#control-panel-template-hooks

The hook renders _inside_ the main tab, but you can create a "new" tab by first closing the Order Details tab `div` and then opening a new one of your own following the same format.

It's a bit of a hack, and a better hook setup would be ideal... but this method has been reliable for me across several cases.
